src/GsvdInitialization.jl

```diff
@@ -176,4 +176,40 @@ function Wcols_modification(X::AbstractArray{T}, W::AbstractArray{T}, H::Abstrac
     return β[:]
 end
 
+function gsvdrecover_2r(X::AbstractArray, W0::AbstractArray, H0::AbstractArray, kadd::Int, f::Tuple)
+    m, n = size(W0)
+    kadd <= n || throw(ArgumentError("# of extra columns must less than 1st NMF components"))
+    if kadd == 0
+        return W0, H0, 0
+    else
+        U0, S0, V0 = f
+        U0, S0, V0 = U0[:,1:n], S0[1:n], V0[:,1:n]
+        Hadd, Λ = init_H(U0, S0, V0, W0, H0, kadd)
+        Wadd, a = init_W(X, W0, H0, Hadd)
+        # @show Wadd, Hadd
+        Wadd_nn, Hadd_nn = init2r(Wadd, Hadd)
+        W0_1, H0_1 = [repeat(a', m, 1).*W0 Wadd_nn], [H0; Hadd_nn]
+        cs = Wcols_modification(X, W0_1, H0_1)
+        # @show cs
+        W0_2, H0_2 = repeat(cs', m, 1).*W0_1, H0_1
+        # W0_2, H0_2 = W0_1, H0_1
+        return abs.(W0_2), abs.(H0_2), Λ
```

Comment:
Why is the `abs` needed? If it has tiny negative values, would `max.(0, W0_2)` be more appropriate? `abs` nominally runs the risk of turning a large negative number positive.